Item updating & removing in list

dronn1kDev opened the pull request.

dronn1kDev committed:
updateItem has been implemented.

dronn1kDev committed:
upate item usage added to the example.

vulpeep commented on lib/src/controller/flutter_pagination_controller.dart:
```diff
@@ -93,4 +86,72 @@ class FlutterPaginationController<ItemType, PM extends PaginationMethod,
+  Future<void> getFirst() async {
+    isProcessing.value = true;
+    _state = await handlePagination(firstPagePointer, true);
+    isProcessing.value = false;
```
There is no error handling, isProcessing will hang at true if an exception occurs in the asynchronous call

dronn1kDev committed:
try finally has been added around handle method.

dronn1kDev committed:
some improvements.

CallbackDepthProcessor implemented.
updateItem & RemoveItemAt implemented for the PaginationController.

dronn1kDev committed:
Minor fixes.

CallbackDepthProcessor fixed.
example fixes.
Documentation is provided for new methods.

dronn1kDev committed:
Example was renamed.

vulpeep commented on lib/src/controller/flutter_pagination_controller.dart:
```diff
@@ -1,5 +1,6 @@
 import 'package:flutter/rendering.dart';
+import 'package:pagination_controller/src/base/callback_depth_processor.dart';
 
 import '../base/pagination_controller_base.dart';
```
Bring the imports to a consistent style

vulpeep reviewed (commented):
And fix the example project name (simple_exmaple) in the code, not just in the path

dronn1kDev committed:
MR fixes.

Imports were fixed.
Example name was fixed.

dronn1kDev committed:
dependencies was updated.
